File: lib/domain/entities/project.dart
enum ProjectType { general, chat, code, app, research, content }

enum ProjectStatus { active, archived, deleted }

class Project {
  final String id;
  final String name;
  final String? description;
  final ProjectType type;
  final ProjectStatus status;
  final Map<String, dynamic> aiContext;
  final List<String> fileIds;
  final List<String> conversationIds;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Project({
    required this.id,
    required this.name,
    this.description,
    this.type = ProjectType.general,
    this.status = ProjectStatus.active,
    this.aiContext = const {},
    this.fileIds = const [],
    this.conversationIds = const [],
    required this.createdAt,
    required this.updatedAt,
  });

}

File: backend/src/models/project.py

from sqlalchemy import String, Boolean, DateTime, Text, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column
from sqlalchemy.sql import func
from datetime import datetime
import uuid
from ..core.database import Base

class Project(Base):
    __tablename__ = "projects"
    
    id: Mapped[str] = mapped_column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id: Mapped[str] = mapped_column(String(36), ForeignKey("users.id", ondelete="CASCADE"), index=True, nullable=False)
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    description: Mapped[str] = mapped_column(Text, nullable=True)
    type: Mapped[str] = mapped_column(String(50), default="general")
    status: Mapped[str] = mapped_column(String(20), default="active")
    ai_context: Mapped[str] = mapped_column(Text, default="{}")
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
